flecs_ecs/src/core/utility/traits/option_or_zst.rs

```diff
@@ -0,0 +1,33 @@
+/// Utility trait for `try_get` callbacks
+///
+/// If the callback returns `Option`, pass it through.
+/// If the callback returns `()`, convert the `try_get` status into a `bool` instead.
+pub trait OptionOrZST {
```

Comment:
I know naming is difficult, but any chance we can come up with a better name? initially when I was looking at try_get, I thought this trait meant "you can only `get` options or zst's".

Reply:
Resolved by refactoring so the trait isn't needed anymore